.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-json
+      - id: check-added-large-files
+      - id: check-merge-conflict
```

Comment:
are these hooks fully compatible with what we are already formatting via github actions workflow and `make check-python/check-rust` commands? if not, we need to make them behave the same. And if possible, reuse the same `make` commands for reusability